packages/legend-query-builder/src/stores/QueryBuilderState.ts

```diff
@@ -444,12 +461,42 @@ export abstract class QueryBuilderState implements CommandRegistrar {
     return PrimitiveType.STRING;
   }
 
-  initializeWithQuery(query: RawLambda): void {
-    this.rebuildWithQuery(query);
+  initializeWithQuery(
+    query: RawLambda,
+    defaultParameterValues?: Map<string, ValueSpecification>,
+  ): void {
+    this.rebuildWithQuery(query, {
+      defaultParameterValues,
+    });
     this.resetQueryResult();
     this.changeDetectionState.initialize(query);
   }
 
+  buildInitialParameterValues(
+    preserveParameterValues: boolean | undefined,
+    defaultParameterValues: Map<string, ValueSpecification> | undefined,
+  ): Map<string, QueryBuilderParameterValue> | undefined {
+    const res = new Map<string, QueryBuilderParameterValue>();
+    if (defaultParameterValues?.size) {
+      Array.from(defaultParameterValues.entries()).forEach(([k, v]) => {
+        res.set(k, {
+          variable: k,
+          value: v,
+        });
+      });
+      return res;
+    } else if (preserveParameterValues) {
```

Comment:
I would think `preserveParameterValues` take higher precedence than `defaultParameterValues`, here are we saying, for example I'm running my query A
- A has default param: x = 1
- We run A set x = 2
- We go to text-mode, when we come back to form mode, x = 1

Reply:
Fair point. 
My assumption was that either or would be set but not both. Will change